packages/components/organisms/f-checkout/src/components/tests/Checkout.test.js

```diff
@@ -601,6 +612,95 @@ describe('Checkout', () => {
                     expect(wrapper.vm.$v.fulfilment.address.postcode).toBeDefined();
                 });
             });
+
+            describe('if isLoggedIn set to false', () => {
+                afterEach(() => {
+                    jest.clearAllMocks();
+                });
+
+                it('should call `setupGuestUser`', async () => {
+                    const state = {
+                        ...defaultState,
+                        isLoggedIn: false
+                    };
+
+                    const setupGuestUserSpy = jest.spyOn(VueCheckout.methods, 'setupGuestUser');
+
+                    const wrapper = shallowMount(VueCheckout, {
+                        store: createStore(state),
+                        i18n,
+                        localVue,
+                        propsData
+                    });
+
+                    await wrapper.vm.onFormSubmit();
+
+                    expect(setupGuestUserSpy).toHaveBeenCalled();
+                });
+
+                it('should call `createGuestUser`', async () => {
```

Comment:
This test, in my humble opinion, should be placed under the `setupGuestUser` method tests, rather than `submitCheckout`.